### PR TITLE
Adds new operations to pmc workload

### PR DESCRIPTION
**Pull request by harshavamsi**
Adds new Operations to the PMC workload to support nested prefix queries

Signed-off-by: Harsha Vamsi Kalluri <hvamsi@amazon.com>

### CODE REVIEW

**Review by IanHoang — COMMENTED** [review]
@harshavamsi As standard procedure, could you run the PMC workload with these latest changes in `--test-mode` and provide the sample of the output in the PR description? This will ensure that there are no issues before merging this in.

**Comment by harshavamsi** [comment]
@IanHoang here's a sample run from the new test procedure.

```
opensearch-benchmark-workloads git:(main) opensearch-benchmark execute-test --pipeline=benchmark-only --workload=pmc --target-host="opense-clust-JeMftrsCHhzL-4e64d68df0dff2e6.elb.us-west-2.amazonaws.com:80"  --workload-repository="/home/hvamsi/code/opensearch-benchmark-workloads" --kill-running-processes --test-procedure="append-and-prefix" --test-mode

   ____                  _____                      __       ____                  __                         __
  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
 / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
    /_/

[INFO] Executing test with workload [pmc], test_procedure [append-and-prefix] and provision_config_instance ['external'] with version [3.0.0-SNAPSHOT].

[WARNING] merges_total_time is 4280354 ms indicating that the cluster is not in a defined clean state. Recorded index time metrics may be misleading.
[WARNING] merges_total_throttled_time is 167378 ms indicating that the cluster is not in a defined clean state. Recorded index time metrics may be misleading.
[WARNING] indexing_total_time is 13910583 ms indicating that the cluster is not in a defined clean state. Recorded index time metrics may be misleading.
[WARNING] refresh_total_time is 407720 ms indicating that the cluster is not in a defined clean state. Recorded index time metrics may be misleading.
[WARNING] flush_total_time is 432345 ms indicating that the cluster is not in a defined clean state. Recorded index time metrics may be misleading.
Running put-settings                                                           [100% done]
Running delete-index                                                           [100% done]
Running create-index                                                           [100% done]
Running check-cluster-health                                                   [100% done]
Running index-append                                                           [100% done]
Running refresh-after-index                                                    [100% done]
Running force-merge                                                            [100% done]
Running refresh-after-force-merge                                              [100% done]
Running wait-until-merges-finish                                               [100% done]
Running prefix_index                                                           [100% done]
Running prefix_docvalues                                                       [100% done]
Running prefix_indexordocvalues                                                [100% done]
Running nested_prefix_index                                                    [100% done]
Running nested_prefix_docvalues                                                [100% done]
Running nested_prefix_indexordocvalues                                         [100% done]

------------------------------------------------------
    _______             __   _____
   / ____(_)___  ____ _/ /  / ___/_________  ________
  / /_  / / __ \/ __ `/ /   \__ \/ ___/ __ \/ ___/ _ \
 / __/ / / / / / /_/ / /   ___/ / /__/ /_/ / /  /  __/
/_/   /_/_/ /_/\__,_/_/   /____/\___/\____/_/   \___/
------------------------------------------------------
            
|                                                         Metric |                           Task |       Value |   Unit |
|---------------------------------------------------------------:|-------------------------------:|------------:|-------:|
|                     Cumulative indexing time of primary shards |                                |     231.846 |    min |
|             Min cumulative indexing time across primary shards |                                |     0.01165 |    min |
|          Median cumulative indexing time across primary shards |                                |     0.01255 |    min |
|             Max cumulative indexing time across primary shards |                                |     231.784 |    min |
|            Cumulative indexing throttle time of primary shards |                                |           0 |    min |
|    Min cumulative indexing throttle time across primary shards |                                |           0 |    min |
| Median cumulative indexing throttle time across primary shards |                                |           0 |    min |
|    Max cumulative indexing throttle time across primary shards |                                |           0 |    min |
|                        Cumulative merge time of primary shards |                                |     71.3392 |    min |
|                       Cumulative merge count of primary shards |                                |         105 |        |
|                Min cumulative merge time across primary shards |                                |           0 |    min |
|             Median cumulative merge time across primary shards |                                |           0 |    min |
|                Max cumulative merge time across primary shards |                                |     71.3392 |    min |
|               Cumulative merge throttle time of primary shards |                                |     2.78963 |    min |
|       Min cumulative merge throttle time across primary shards |                                |           0 |    min |
|    Median cumulative merge throttle time across primary shards |                                |           0 |    min |
|       Max cumulative merge throttle time across primary shards |                                |     2.78963 |    min |
|                      Cumulative refresh time of primary shards |                                |     6.79487 |    min |
|                     Cumulative refresh count of primary shards |                                |         153 |        |
|              Min cumulative refresh time across primary shards |                                |       0.003 |    min |
|           Median cumulative refresh time across primary shards |                                |  0.00325833 |    min |
|              Max cumulative refresh time across primary shards |                                |     6.77888 |    min |
|                        Cumulative flush time of primary shards |                                |     7.20575 |    min |
|                       Cumulative flush count of primary shards |                                |          32 |        |
|                Min cumulative flush time across primary shards |                                |           0 |    min |
|             Median cumulative flush time across primary shards |                                |           0 |    min |
|                Max cumulative flush time across primary shards |                                |     7.20575 |    min |
|                                        Total Young Gen GC time |                                |       0.008 |      s |
|                                       Total Young Gen GC count |                                |           1 |        |
|                                          Total Old Gen GC time |                                |           0 |      s |
|                                         Total Old Gen GC count |                                |           0 |        |
|                                                     Store size |                                |     24.2149 |     GB |
|                                                  Translog size |                                | 3.07336e-07 |     GB |
|                                         Heap used for segments |                                |           0 |     MB |
|                                       Heap used for doc values |                                |           0 |     MB |
|                                            Heap used for terms |                                |           0 |     MB |
|                                            Heap used for norms |                                |           0 |     MB |
|                                           Heap used for points |                                |           0 |     MB |
|                                    Heap used for stored fields |                                |           0 |     MB |
|                                                  Segment count |                                |          42 |        |
|                                                 Min Throughput |                   index-append |     2186.85 | docs/s |
|                                                Mean Throughput |                   index-append |     2186.85 | docs/s |
|                                              Median Throughput |                   index-append |     2186.85 | docs/s |
|                                                 Max Throughput |                   index-append |     2186.85 | docs/s |
|                                        50th percentile latency |                   index-append |     364.527 |     ms |
|                                       100th percentile latency |                   index-append |      554.14 |     ms |
|                                   50th percentile service time |                   index-append |     364.527 |     ms |
|                                  100th percentile service time |                   index-append |      554.14 |     ms |
|                                                     error rate |                   index-append |           0 |      % |
|                                                 Min Throughput |       wait-until-merges-finish |      148.67 |  ops/s |
|                                                Mean Throughput |       wait-until-merges-finish |      148.67 |  ops/s |
|                                              Median Throughput |       wait-until-merges-finish |      148.67 |  ops/s |
|                                                 Max Throughput |       wait-until-merges-finish |      148.67 |  ops/s |
|                                       100th percentile latency |       wait-until-merges-finish |     6.35279 |     ms |
|                                  100th percentile service time |       wait-until-merges-finish |     6.35279 |     ms |
|                                                     error rate |       wait-until-merges-finish |           0 |      % |
|                                                 Min Throughput |                   prefix_index |      165.25 |  ops/s |
|                                                Mean Throughput |                   prefix_index |      165.25 |  ops/s |
|                                              Median Throughput |                   prefix_index |      165.25 |  ops/s |
|                                                 Max Throughput |                   prefix_index |      165.25 |  ops/s |
|                                       100th percentile latency |                   prefix_index |     7.82741 |     ms |
|                                  100th percentile service time |                   prefix_index |     1.56098 |     ms |
|                                                     error rate |                   prefix_index |           0 |      % |
|                                                 Min Throughput |               prefix_docvalues |      145.32 |  ops/s |
|                                                Mean Throughput |               prefix_docvalues |      145.32 |  ops/s |
|                                              Median Throughput |               prefix_docvalues |      145.32 |  ops/s |
|                                                 Max Throughput |               prefix_docvalues |      145.32 |  ops/s |
|                                       100th percentile latency |               prefix_docvalues |     8.44432 |     ms |
|                                  100th percentile service time |               prefix_docvalues |     1.35699 |     ms |
|                                                     error rate |               prefix_docvalues |           0 |      % |
|                                                 Min Throughput |        prefix_indexordocvalues |      126.34 |  ops/s |
|                                                Mean Throughput |        prefix_indexordocvalues |      126.34 |  ops/s |
|                                              Median Throughput |        prefix_indexordocvalues |      126.34 |  ops/s |
|                                                 Max Throughput |        prefix_indexordocvalues |      126.34 |  ops/s |
|                                       100th percentile latency |        prefix_indexordocvalues |     11.0342 |     ms |
|                                  100th percentile service time |        prefix_indexordocvalues |     2.91456 |     ms |
|                                                     error rate |        prefix_indexordocvalues |           0 |      % |
|                                                 Min Throughput |            nested_prefix_index |      147.29 |  ops/s |
|                                                Mean Throughput |            nested_prefix_index |      147.29 |  ops/s |
|                                              Median Throughput |            nested_prefix_index |      147.29 |  ops/s |
|                                                 Max Throughput |            nested_prefix_index |      147.29 |  ops/s |
|                                       100th percentile latency |            nested_prefix_index |     8.40943 |     ms |
|                                  100th percentile service time |            nested_prefix_index |      1.3955 |     ms |
|                                                     error rate |            nested_prefix_index |           0 |      % |
|                                                 Min Throughput |        nested_prefix_docvalues |      135.96 |  ops/s |
|                                                Mean Throughput |        nested_prefix_docvalues |      135.96 |  ops/s |
|                                              Median Throughput |        nested_prefix_docvalues |      135.96 |  ops/s |
|                                                 Max Throughput |        nested_prefix_docvalues |      135.96 |  ops/s |
|                                       100th percentile latency |        nested_prefix_docvalues |     9.05091 |     ms |
|                                  100th percentile service time |        nested_prefix_docvalues |      1.4427 |     ms |
|                                                     error rate |        nested_prefix_docvalues |           0 |      % |
|                                                 Min Throughput | nested_prefix_indexordocvalues |      120.23 |  ops/s |
|                                                Mean Throughput | nested_prefix_indexordocvalues |      120.23 |  ops/s |
|                                              Median Throughput | nested_prefix_indexordocvalues |      120.23 |  ops/s |
|                                                 Max Throughput | nested_prefix_indexordocvalues |      120.23 |  ops/s |
|                                       100th percentile latency | nested_prefix_indexordocvalues |      11.355 |     ms |
|                                  100th percentile service time | nested_prefix_indexordocvalues |     2.80758 |     ms |
|                                                     error rate | nested_prefix_indexordocvalues |           0 |      % |


--------------------------------
[INFO] SUCCESS (took 13 seconds)
--------------------------------
```

**Comment by harshavamsi** [comment]
Hi @IanHoang can we merge this so that it can be picked up by the nightly benchmarking?

File: pmc/index.json [hunk 1]
```diff
@@ -11,7 +11,20 @@
     "dynamic": "strict",
     "properties": {
       "name": {
-        "type": "keyword"
+        "type": "keyword",
+        "index": true,
+        "doc_values": false,
+        "copy_to": ["name_docvalues", "name_indexordocvalues"]
```
Comment (gkamat) [inline pmc/index.json:17]: What is the reason the default for `doc_values` being changed for the `name` field?  Any additions to the workload should not modify the existing mapping.

**Review by gkamat — CHANGES_REQUESTED** [review]

**Comment by IanHoang** [comment]
Spoke to @harshavamsi and am closing this out

**Pull request closed without merging**